src/settings.c

```diff
@@ -171,14 +177,19 @@ int git_libgit2_opts(int key, ...)
 		{
 			const char *file = va_arg(ap, const char *);
 			const char *path = va_arg(ap, const char *);
-			if (!SSL_CTX_load_verify_locations(git__ssl_ctx, file, path)) {
-				giterr_set(GITERR_NET, "SSL error: %s",
-					ERR_error_string(ERR_get_error(), NULL));
-				error = -1;
-			}
+			error = git_openssl_set_cert_file(file, path);
+		}
+#elif GIT_MBEDTLS
+		{
+			const char *file = va_arg(ap, const char *);
+			const char *path = va_arg(ap, const char *);
+			if (file)
+				error = git_mbedtls_set_cert_file(file, 0);
+			if (error && path)
+				error = git_mbedtls_set_cert_file(path, 0);
```

Comment:
Since we want to support passing in both a file and a path at the same time the following should work:
```
if (file || path) {
	error = -1;
	if (file)
		error = git_mbedtls_set_cert_file(file, 0);
	if (error < 0 && path)
		error = git_mbedtls_set_cert_file(path, 1);
} else {
	giterr_set(GITERR_NET, "cannot set certificate locations: no file or path given");
	error = -1;
}
```
I think this will have to be handled in libgit2 as mbedTLS only has [`mbedtls_x509_crt_parse_file`](https://tls.mbed.org/api/group__x509__module.html#gad4da63133d3590aa311488497d4c38ec) and [`mbedtls_x509_crt_parse_path`](https://tls.mbed.org/api/group__x509__module.html#ga571fc89b9f3217ab3dd67bd7af905066).